pkg/reader/usdc_reader.go

```diff
@@ -120,14 +129,34 @@ func (u usdcMessageReader) MessageHashes(
 		return nil, fmt.Errorf("no contract bound for chain %d", source)
 	}
 
+	eventFilter := make([]query.Expression, 0, len(eventIDs))
+	for _, id := range eventIDs {
+		eventFilter = append(
+			eventFilter,
+			query.Comparator(consts.EventFilterCCIPMessageSent,
+				primitives.ValueComparator{
+					Value:    id,
+					Operator: primitives.Eq,
+				}),
+		)
+	}
+
+	filter := []query.Expression{
+		query.Or(eventFilter...),
+		query.Confidence(primitives.Finalized),
+	}
+
```

Comment:
You can format the key filter with query.Where
```suggestion
		keyFilter, err := query.Where(consts.EventNameCCTPMessageSent,
		query.Or(eventFilter...),
		query.Confidence(primitives.Finalized))
	if err != nil {
		return nil, err
	}

	iter, err := u.contractReaders[source].QueryKey(
		ctx,
		cr,
		keyFilter,
```